haystack/nodes/reader/farm.py

```diff
@@ -1016,6 +1019,10 @@ def _extract_answers_of_predictions(self, predictions: List[QAPred], top_k: Opti
         answers = sorted(answers, reverse=True)
         answers = answers[:top_k]
 
+        # apply confidence based filtering if enabled
+        if self.confidence_threshold is not None:
+            answers = [x for x in answers if x.score >= self.confidence_threshold]  # type: ignore
```

Comment:
We try to avoid `# type: ignore` as much as possible, so it would be good to remove this. I think this line would be fine for mypy if we change it to `answers = [ans for ans in answers if ans.score is not None and ans.score >= self.confidence_threshold]`

Reply:
Thanks Bogdan  : )